jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildConfiguration.java

```diff
@@ -178,6 +188,69 @@ public BuildConfiguration build() {
           throw new IllegalStateException(errorMessage.toString());
       }
     }
+
+    /**
+     * Converts/validates a list of ports with ranges to an expanded form without ranges.
+     *
+     * <p>Example: ["1000/tcp", "2000-2002/tcp"] -> ["1000/tcp", "2000/tcp", "2001/tcp", "2002/tcp"]
+     *
+     * @param ports the list of port numbers/ranges
+     * @return the ports as a list of integers
+     * @throws NumberFormatException if any of the ports are in an invalid format or out of range
+     */
+    @VisibleForTesting
+    ImmutableList<String> expandPortRanges(List<String> ports) throws NumberFormatException {
+      ImmutableList.Builder<String> result = new ImmutableList.Builder<>();
+
+      for (String port : ports) {
+        // Make sure configuration is a single number or a range, with an optional protocol
+        // Examples:
+        //   100
+        //   200-210
+        //   1000/tcp
+        //   2000/udp
+        //   500-600/tcp
+        if (!port.matches("^\\d+(-\\d+)?(\\/tcp|\\/udp)?$")) {
+          throw new NumberFormatException(
+              "Invalid port configuration: '"
+                  + port
+                  + "'. Make sure the port is a single number or a range of two numbers separated "
+                  + "with a '-', with or without protocol specified (e.g. '<portNum>/tcp' or "
+                  + "'<portNum>/udp').");
+        }
+
+        // Parse protocol
+        List<String> splitProtocol = Splitter.on("/").splitToList(port);
+        String protocol = splitProtocol.size() > 1 ? "/" + splitProtocol.get(1) : "";
+
+        // Parse range (or treat as range of min-min if only single port configuration)
+        List<String> splitRange = Splitter.on("-").splitToList(splitProtocol.get(0));
+        int min = Integer.parseInt(splitRange.get(0));
+        int max = min;
+        if (splitRange.size() > 1) {
+          max = Integer.parseInt(splitRange.get(1));
+        }
+
+        // Error if configured as 'max-min' instead of 'min-max'
+        if (min > max) {
+          throw new NumberFormatException(
+              "Invalid port range '" + port + "'; smaller number must come first.");
+        }
+
+        // Warn for possibly invalid port numbers
+        if (min < 1 || max > 65535) {
+          // TODO: Add details/use HelpfulSuggestions for these warnings
+          buildLogger.warn("Port number '" + port + "' is out of usual range (1-65535).");
+        }
+
+        // Add all numbers in range to list
+        for (int portNum = min; portNum <= max; portNum++) {
+          result.add(portNum + protocol);
```

Comment:
Since we are passing this through the code, we might want to consider putting this in a canonical form rather than the docker image format for of `port/protocol`. Like a class that has a port and a protocol. Can also just be in a TODO for now.